python/dgl/sampling/neighbor.py

```diff
@@ -155,8 +181,21 @@ def sample_neighbors(g, nodes, fanout, edge_dir='in', prob=None, replace=False,
             else:
                 prob_arrays.append(nd.array([], ctx=nd.cpu()))
 
+    excluded_edges_all_t = []
+    if exclude_edges is not None:
+        if not isinstance(exclude_edges, dict):
+            if len(g.etypes) > 1:
+                raise DGLError("Must specify etype type when the graph is not homogeneous.")
+            exclude_edges = {g.canonical_etypes[0] : exclude_edges}
+        exclude_edges = utils.prepare_tensor_dict(g, exclude_edges, 'edges')
+        for etype in g.canonical_etypes:
+            if etype in exclude_edges:
+                excluded_edges_all_t.append(F.to_dgl_nd(exclude_edges[etype]))
+            else:
+                excluded_edges_all_t.append(nd.array([], ctx=nd.cpu()))
```

Comment:
does it generate a right list of excluded edges? the list should be indexed by edge type IDs, but you iterate over the edge type based its order in the dict. There is no guarantee that the order is the same as indicated by edge type IDs.

Reply:
Is this resolved?  I saw it is now iterating over `g.canonical_etypes`.